### PR TITLE
Add office31 for Deep models

### PR DESCRIPTION
**Pull request by YanisLalou**
Old office31 with preprocessing, now called: `Office31Decaf`
Office31 for deep models, now called: `Office31`

### CODE REVIEW

File: datasets/office31.py [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
-    from skada.datasets import fetch_office31_decaf_all, Office31CategoriesPreset
+    import torchvision.transforms as transforms
+    import os
+    from torch.utils.data import Dataset, DataLoader
+    from PIL import Image
+    from pathlib import Path
+    import sys
+
+    PATH_skada_bench = Path(__file__).resolve().parents[1]
+    sys.path.extend([str(PATH_skada_bench)])
```
Comment (antoinecollas) [inline datasets/office31.py:14]: Why do you need this? it's ugly as hell...

File: datasets/office31.py [hunk 2]
```diff
@@ -25,33 +33,45 @@ class Dataset(BaseDataset):
...
+    url_dataset = "https://wjdcloud.blob.core.windows.net/dataset/OFFICE31.zip"
+
+
+    def _get_dataset(self, domain_select):
+        # Define transformations to preprocess the images
+        preprocess = transforms.Compose([
+            transforms.Resize(256),
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
```
Comment (antoinecollas) [inline datasets/office31.py:49]: where did you get these values?

Comment (YanisLalou) [reply datasets/office31.py:49]: imagenet normalisation terms since we'll be using a pertained resnet50

**Review by antoinecollas — APPROVED** [review]
LGTM
